semgrep/semgrep/output.py

```diff
@@ -92,6 +92,9 @@ def build_normal_output(
                     severity_prepend = f"{severity} "
             yield f"{severity_prepend}{YELLOW_COLOR}rule:{check_id}: {message}{RESET_COLOR}"
 
+        if extra.get("contents"):
```

Comment:
`pattern-regex` operates on the entire file buffer, not line-by-line. This was intentionally done so that it can find code like:

```python
some_func_call(
    "INSECURE_STRING"
)
```

Operating line-by-line wouldn't allow for an expression to consider the argument above. The downside is that we don't have an easy way to access line and column information (it's all just one big file contents string). This means we couldn't leverage existing `["extra"]["lines"]` functionality. We instead store the regex match contents directly in the finding and reference it here as `contents`.

Reply:
re: decision to do whole buffer instead of line by line -> makes sense. Having a separate result format adds a fair bit of complexity. Let me think about this for a bit.

Reply:
Probably fine for now. We can iterate iterate :D